index.bs

```diff
@@ -575,31 +539,52 @@ To <dfn>handle an incoming message</dfn> given a [=WebSocket connection=]
     Issue: Nothing seems to define what [=status codes|status code=]
     is used for UTF-8 errors.
 
- 1. Let |result| be the result of [=process a command|processing a
-    command=] given |data|.
+ 1. Match |data| against the [=remote end definition=]. If this results in a
+    match:
 
- 1. If |result| is an [=error=], then [=respond with an error=] given
-    |connection|, |data|, and |result|'s [=error code=], and finally return.
+    1. Let |parsed| be the map representing the matched data.
 
- 1. Assert: |result| is a [=success=]
+    1. Assert: |parsed| contains "<code>id</code>", "method", and
+       "<code>params</code>"
 
- 1. Let |command id| and |value| be the two fields |result|'s data.
+    1. Let |command id| be |parsed|["<code>id</code>"].
 
- 1. Let |response| be a new [=map=] with the following properties:
+    1. Let |method| be |parsed|["<code>method</code>"]
 
-   <dl>
-      <dt>"id"</dt>
-      <dd>|command id|</dd>
+    1. Run the following steps in parallel:
```

Comment:
How to run the steps in parallel? They seem to have dependency on the previous step.

Reply:
Looks like the definition of "parallel" here is borrowed from the HTML spec: https://html.spec.whatwg.org/multipage/infrastructure.html#parallelism

The substeps under "Run the following steps in parallel" start after the previous steps have completed. The substeps are run in order one at a time, but they don't block the main "handle an incoming message" algorithm, so it is free to handle more messages.